source/agora/common/crypto/Key.d

```diff
@@ -190,6 +191,24 @@ public struct PublicKey
         dg(this.data[]);
     }
 
+    /***************************************************************************
+
+        PublicKey fromBinary
+
+        Params:
+            dg = deserialize function
+
+        Returns:
+            `PublicKey` Public key address
+
+    ***************************************************************************/
+
+    public static PublicKey fromBinary (scope DeserializeDg dg) nothrow @safe
+    {
+        alias DType = typeof(this.data);
+        return PublicKey(DType(dg(DType.Width)));
```

Comment:
```suggestion
        return PublicKey(DType(dg(DType.sizeof)));
```

Reply:
@Geod24 That's what he told me.

Reply:
Well it shouldn't really make a difference, except `.sizeof` can always be trusted. But here `.sizeof == .Width`, so use whichever one you prefer.

Reply:
`.sizeof` is better ;)

Reply:
I use .sizeof